backend/runner/taskrun/schedulerv2.go

```diff
@@ -109,35 +118,68 @@ func (s *SchedulerV2) scheduleAutoRolloutTasks(ctx context.Context) error {
 		return errors.Wrapf(err, "failed to list environments")
 	}
 
-	var autoRolloutEnvironmentIDs []int
 	for _, environment := range environments {
 		policy, err := s.store.GetRolloutPolicy(ctx, environment.UID)
 		if err != nil {
-			return errors.Wrapf(err, "failed to get rollout policy for environment ID %d", environment.UID)
+			return errors.Wrapf(err, "failed to get rollout policy for environment %d", environment.UID)
 		}
-		if !policy.Automatic {
-			continue
+
+		taskIDs, err := s.store.ListTasksToAutoRollout(ctx, []int{environment.UID})
```

Comment:
We need to check if the task can auto rollout:

1. If the rollout policy is automatic
2. If users don't have a correct subscription plan
3. If the task's instance doesn't have the license.

I'm not sure about the performance issue, PTAL @RainbowDashy

Reply:
worse but negligible.
Not sure why you moved `ListTasksToAutoRollout` into the loop though. We can get them in one query.

Reply:
We can list all task ids in outside, but we still need to filter them by environment id in the loop

Reply:
You can pass environmentIDs to `ListTasksToAutoRollout` which then uses them to filter.